q2cli/builtin/tools.py

```diff
@@ -80,40 +80,149 @@ def export_data(input_path, output_path, output_format):
     click.echo(CONFIG.cfg_style('success', success))
 
 
-def show_importable_types(ctx, param, value):
-    if not value or ctx.resilient_parsing:
-        return
+def print_descriptions(descriptions, tsv):
+    if tsv:
+        for value, description in descriptions.items():
+            click.echo(f"{value}\t", nl=False)
+            if description:
+                click.echo(deformat_description(description))
+            else:
+                click.echo()
+    else:
+        import textwrap
+        for value, description in descriptions.items():
+            click.secho(value, bold=True)
+            if description:
+                description = deformat_description(description)
+                tabsize = 8
+                wrapped_description = textwrap.wrap(description,
+                                                    width=72-tabsize,
+                                                    initial_indent='\t',
+                                                    subsequent_indent='\t',
+                                                    tabsize=tabsize)
+                for line in wrapped_description:
+                    click.echo(f"{line}")
+            else:
+                click.secho("\tNo description", italic=True)
+            click.echo()
+
+
+def deformat_description(description):
+    import re
+    deformatted = re.sub(r"[\t\n]+", ' ', description)
+    despaced = re.sub(r" +", ' ', deformatted)
+    return despaced
+
+
+def get_matches(words, possibilities, cutoff=0.5):
+    from difflib import get_close_matches
+    matches = []
+    for word in words:
+        matches += get_close_matches(word,
+                                     possibilities,
+                                     n=len(possibilities),
+                                     cutoff=cutoff)
+        # substring search
```

Comment:
Is this substring search not already handled by difflib? I notice that my search example above for `mpledata` matches to `SampleData[JoinedSequencesWithQuality]`, which is a short substring for the type being matched.

Reply:
No it isn't, only overall similarity based.

Reply:
It may match because of the explicit substring search, or they may be similar enough